package/upgrade/upgrade_node.sh

```diff
@@ -294,6 +301,53 @@ clean_rke2_archives() {
     done
 }
 
+convert_nodenetwork_to_vlanconfig() {
+  [[ $UPGRADE_PREVIOUS_VERSION != "1.0.3" ]] && return
+
+  # Don't need to convert nodenetwork if harvester-mgmt is used as vlan interface in any nodenetwork
+  [[ $(kubectl get nodenetwork -o yaml | yq '.items[].spec.nic | select(. == "harvester-mgmt")') ]] && return
+
+  local name=${HARVESTER_UPGRADE_NODE_NAME}-vlan
+
+  nn=$(kubectl get nn "$name" -o yaml)
+  vlan_interface=$(echo "$nn" | yq '.spec.nic')
+
+  [[ "$vlan_interface" == "null" ]] && echo "Invalid nodenetwork $name" && return
+
+  type=$(echo "$nn" | yq e '.spec.nic as $nic | .status.networkLinkStatus.[$nic].type')
+  if [ "$type" == "bond" ]; then
+    vlan_interface_details=$(v="$vlan_interface" yq '.install.networks.[env(v)]' /oem/harvester.config)
+    nics=($(echo "$vlan_interface_details" | yq '.interfaces[].name'))
+    mtu=$(echo "$vlan_interface_details" | yq '.mtu')
+    mode=$(echo "$vlan_interface_details" | yq '.bondoptions.mode')
+    miimon=$(echo "$vlan_interface_details" | yq '.bondoptions.miimon')
+  else
+    nics=("$vlan_interface")
+    mtu=0
+    mode="active-backup"
+    miimon=0
+  fi
+
+  echo "
```

Comment:
such writing is difficult to debug, when there are many to-be-replaced variables

please consider cat the input into a temp file, then cat the file, finally call kubectl

Reply:
From my perspective, kubectl command will return enough error information if there is an error.